packages/CollapsibleChecklists/src/components/Group/Group.js

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Collapsible from "@paprika/collapsible";
+import textContent from "react-addons-text-content";
+import Item from "../Item";
+import groupStyles from "./Group.styles";
+
+const propTypes = {
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]), // are probably an array of "Items", but could be a Spinner or anything else
+  isDisabled: PropTypes.bool,
+  onChange: PropTypes.func,
+  onExpand: PropTypes.func,
+  title: PropTypes.node.isRequired,
+};
+
+const defaultProps = {
+  children: [],
+  isDisabled: false,
+  onChange: () => {},
+  onExpand: () => {},
+};
+
+function useIsIndeterminate(checkboxRef) {
+  const [, setIsIndeterminate] = React.useState(null);
+  React.useEffect(() => {
+    setIsIndeterminate(checkboxRef.current.indeterminate);
+  }, [checkboxRef.current.indeterminate]);
+}
+
+function Group(props) {
+  const { children, isDisabled, onChange, onExpand, title } = props;
+  const [isCollapsed, setIsCollapsed] = React.useState(true);
+  const checkboxRef = React.useRef({});
+  useIsIndeterminate(checkboxRef);
+  let allAreChecked = React.Children.count(props.children) > 0;
+  let noneAreChecked = true;
+
+  React.Children.forEach(children, child => {
+    if (child.props.isChecked) {
+      noneAreChecked = false;
+    } else {
+      allAreChecked = false;
+    }
+  });
+
+  if (allAreChecked || noneAreChecked) {
+    checkboxRef.current.indeterminate = false;
+  } else if (!allAreChecked && !noneAreChecked) {
+    checkboxRef.current.indeterminate = true;
+  }
+
+  function toggleChildren() {
+    const childItems =
+      children.length > 0 ? children.filter(childItem => childItem.type.displayName === Item.displayName) : [];
+
+    if (childItems.length === 0) {
+      return;
+    }
+
+    const allChildItemsAreChecked =
+      childItems.filter(childItem => childItem.props.isChecked || childItem.props.isDisabled).length ===
+      childItems.length;
+
+    let childItemsToChange = [];
+    if (allChildItemsAreChecked) {
+      childItemsToChange = childItems.filter(childItem => !childItem.props.isDisabled);
+    } else {
+      childItemsToChange = childItems.filter(childItem => !childItem.props.isChecked && !childItem.props.isDisabled);
+    }
+
+    onChange(childItemsToChange);
+  }
+
+  const label = (
+    <React.Fragment>
+      <input
+        ref={checkboxRef}
+        checked={allAreChecked}
+        type="checkbox"
+        disabled={isDisabled}
+        onChange={toggleChildren}
+        aria-label={textContent(title)}
+      />
+      {title}
```

Comment:
Well, that's what Jamie's trying to do with `textContent(title)`, but as mentioned, that won't really work without rendering the node. Even the library Jamie is using will probably not work for us in all situations:

https://github.com/rwu823/react-addons-text-content/issues/1#issuecomment-236906494

So yeah, we should probably require a string version from the consumer as an `a11yText` prop.

Reply:
... or ... maybe if there was a `visuallyHidden` `<span id={someUID}>{title}</span>`, and then we could have an `aria-labelledby={someUID}` on the `input` ... 🤔

Reply:
But that wouldn't satisfy the requirement of checking the checkbox via clicking the label? Do we still want that?

Reply:
Excellent point! In general clicking the label should activate the checkbox since that's such a small click target... it as a big impact on `a11y`!  In this specific case though, I wonder... there's not an easy "undo" to the select all / deselect all operation of this checkbox, and maybe it's better to confine it to the actual checkbox.  It could easily be assumed by the user that the `title` in this case would actually expand the Collapsible, and then they'd be frustrated.  

I think we need a @krismckinnonacl for this one!

Reply:
Agree with all the discussion here, but I think this can be addressed in another version, this is not the part of the API so we could refactor it easily in a bump it.

Can we create an issue with all the followed steps from this review @jamiek-acl and include this one?

Reply:
I would make clicking the label activate that checkbox. For the following reasons:
1. Assuming that the child entries underneath work normally, if we did make that change, we'd be changing the behaviour of an element that looks visually the same, in close proximity to other controls that do the same thing, in a subtle way - I assume that clicking on the titles of the items underneath would check the associated checkbox, right?
2. If the user does in fact assume that clicking the title expands or collapses the section but instead selects all the elements, it's an easy mistake to correct, because they will encounter it BEFORE they select items underneath. It would be atypical for someone to 
1. click the caret, then 
2. select some things, then 
3. click the caret to collapse again then 
4. come back later and click the title instead of where they clicked before to open the elements.

Reply:
1. Yes checking the titles of the children toggles them, too.
2. No it's not easily undone. If some children are selected and i click the parent, that would select all of its children.  If you didn't intend that and wanted to undo, you wouldn't know which ones to de-select.